src/common.speech/DialogServiceAdapter.ts

```diff
@@ -585,4 +566,57 @@ export class DialogServiceAdapter extends ServiceRecognizerBase {
         const ev = new SpeechRecognitionEventArgs(result, offset, this.privRequestSession.sessionId);
         return ev;
     }
+
+    private handleResponseMessage = (responseMessage: SpeechConnectionMessage): void => {
+        // "response" messages can contain either "message" (activity) or "MessageStatus" data. Fire the appropriate
+        // event according to the message type that's specified.
+        const responsePayload = JSON.parse(responseMessage.textBody);
+        switch (responsePayload.messageType.toLowerCase()) {
+            case "message":
+                const responseRequestId = responseMessage.requestId.toUpperCase();
+                const activityPayload: ActivityPayloadResponse = ActivityPayloadResponse.fromJSON(responseMessage.textBody);
+                const turn = this.privTurnStateManager.GetTurn(responseRequestId);
+
+                // update the conversation Id
+                if (activityPayload.conversationId) {
+                    const updateAgentConfig = this.agentConfig.get();
+                    updateAgentConfig.botInfo.conversationId = activityPayload.conversationId;
+                    this.agentConfig.set(updateAgentConfig);
+                }
+
+                const pullAudioOutputStream: PullAudioOutputStreamImpl = turn.processActivityPayload(
+                    activityPayload,
+                    (SpeechSynthesisOutputFormat as any)[this.privDialogServiceConnector.properties.getProperty(PropertyId.SpeechServiceConnection_SynthOutputFormat, undefined)]);
+                const activity = new ActivityReceivedEventArgs(activityPayload.messagePayload, pullAudioOutputStream);
+                if (!!this.privDialogServiceConnector.activityReceived) {
+                    try {
+                        this.privDialogServiceConnector.activityReceived(this.privDialogServiceConnector, activity);
+                        /* tslint:disable:no-empty */
+                    } catch (error) {
+                        // Not going to let errors in the event handler
+                        // trip things up.
+                    }
+                }
+                break;
+
+            case "messagestatus":
+                if (!!this.privDialogServiceConnector.turnStatusReceived) {
+                    try {
+                        this.privDialogServiceConnector.turnStatusReceived(
+                            this.privDialogServiceConnector,
+                            new TurnStatusReceivedEventArgs(responseMessage.textBody));
+                        /* tslint:disable:no-empty */
+                    } catch (error) {
+                        // Not going to let errors in the event handler
+                        // trip things up.
```

Comment:
Why do we do this? Is there some way to give an error some form of logging and then continue on our way?

Reply:
That's a good question--do we have an "error trace" equivalent here? I don't see one in use but the idea is provide customer diagnosis assistance if an unexpected code is coming back. If we have something better than a lint-dodging console.log, I'd love to use it.

Reply:
Perhaps emitting a BackgroundEvent would work here:
```
Events.instance.onEvent(new BackgroundEvent(error));
```

Reply:
Aha, that's exactly what I was looking for! And it's even used in the same file; shame on me for not seeing it. Looked for "log," "trace," and other variations, but never thought to look for "event!"